feature/chaerim step1

C-limlim opened the pull request:
제가 큰 실수를 저질렀습니다... push 할 때 feature/chaerim_step1에만 했어야하는데 제가 바로 대표브랜치에 push했습니다. 혼자 작성하는 브랜치라 대표브랜치는 reset했고 step1은 다시 pr 요청합니다. 번거롭게해서 죄송해요 ㅠㅠ

수정사항
1. 가독성을 위한 함수 이름 수정
- file/controller.py의 check_file_exists 를 has_file로 변경
- file/service.py의 get_tuple을 get_metadata로 변경

2. 오류 fix
- file/contoller.py의 upload_file 함수에서 확장자 추출 방식 변경

3. 환경변수설정
- env파일과 config.py 추가하여 정보와 소스코드 분리

4. 코드 리펙토링
- file/service.py에서 can_access와 get_file의 역할을 get_file에 통합
- db와 파일 I/O 담당 함수를 분리하고 service단에서 처리

Commit:
feat: add files for step1

= upload, delete, download files
- architecture design

Commit:
refactor: renamed and split function responsibilities

- Renamed functions for clarity
- split the responsibilities of upload/delete functions into two part:
  db and file I/O

Commit:
refactor: renamed and combined functions into one

- merge can_access function to get_file for efficiency
- rename get_tuple to get_metadata

Commit:
fix: upload_file function

- doesn't work properly when 2 or more dots in filename
- considered the case that no extention type in the end of filename

Commit:
feat: add .env, config.py for environment variable management

C-limlim commented on database.py:
```diff
@@ -0,0 +1,17 @@
+
+config = Config()
+
+engine = create_engine(
+    config.DATABASE_URL, connect_args={"check_same_thread": False}
+)
```
정보(실섭db주소)와 소스코드 분리 관련 리뷰: env 파일과 config.py 사용했습니다

C-limlim commented on file/controller.py:
```diff
@@ -0,0 +1,57 @@
+    로컬에 파일 저장, 메타데이터 db에 저장
+    '''
+    file_name = file.filename
+    file_extension = file_name.split('.')[-1] if '.' in file_name else ""
+    server_filename = UPLOAD_DIR + uuid.uuid4().hex + '.' + file_extension
```
파일이름에 `.` 2개 이상 있을경우 리뷰: [-1] 사용 및 확장자 없을 때의 고려를 추가하였습니다.

C-limlim commented on file/controller.py:
```diff
@@ -0,0 +1,57 @@
+    파일 존재하지 않으면 에러반환
+    권한이 없으면 다운 불가능 (본인만)
+    '''
+    if not has_file(file_name, user_id, db=db):
```
코드 최적화 리뷰 (can_access 와 get_file에서 같은 일을 하는 비효율성):
get_file에서 권한이 있는지 확인하고 없을경우 None 반환, 아니면 File을 반환합니다.

이 부분을 작성할 때 HTTPExecption을 service에서 raise하는게 좋은 방식인지에 대한 의문이 들어서 service에서는 권한이 없을 경우 None을 반환하고 Controller에서 None을 받아 raise error를 하는데요. 제 생각이 맞는지 피드백 주시면 감사하겠습니다..!

VSFe replied:
service에서 raise 하는 게 좀 더 나을 것 같아요.

C-limlim replied:
[46901ad](https://github.com/gdsc-ys/mount-be/pull/6/commits/46901adf060257b945f9ab3baf25a3f4a960bb4c)

C-limlim commented on file/service.py:
```diff
@@ -0,0 +1,40 @@
+    target = db.query(File).filter(File.user_id == user_id).filter(File.file_name==file_name).first()
+    return target
+
+def save_file(user_file: File, content:bytes, db=Session):
+    save_file_in_server(user_file.server_filename, content)   
+    db.add(user_file)
+    db.commit()
+    
+def delete_file_service(file_name: str, user_id:str, db: Session):
+    target = get_metadata(file_name, user_id, db)
+    db.delete(target)
+    db.commit()
+    delete_file_in_server(target.server_filename)
+    
+def delete_file_in_server(server_filename: str):
+    os.remove(server_filename)
+    
+def save_file_in_server(server_filename:str, content: bytes):
+    with open(server_filename, 'wb') as f:
+        f.write(content)
```
db, 파일 I/O 역할 분리 및 service단에서의 처리 리뷰:
서로 다른 함수로 분리하였고 controller에서 했던 파일 업로드를 service에서 하도록 변경하였습니다.

VSFe reviewed (approved):
수고하셨습니다~

Commit:
refactoring: raise HTTPException error in service.py

- move code of rasing HTTPException in download_file func into
  service/get_file func
